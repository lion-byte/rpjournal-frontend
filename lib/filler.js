/**
 * @param {object} options
 * @param {string} options.text
 * @param {number} [options.height]
 * @param {number} [options.width]
 * @returns {string}
 */
export const fillerImage = options => {
  const { text, height = 480, width = 640 } = options

  return `https://via.placeholder.com/${width}x${height}.png/535353/eeeeee?text=${encodeURI(
    text
  )}`
}

/**
 * @param {object} [options]
 * @param {number} [options.height]
 * @param {number} [options.width]
 * @returns {string}
 */
export const fillerBackgroundImage = (options = {}) => {
  const { height = 480, width = 640 } = options

  return `https://placeimg.com/${width}/${height}/nature`
}
