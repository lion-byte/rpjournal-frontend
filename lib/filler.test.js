import { fillerImage } from './filler'

describe('Filler', () => {
  test('Create filler image URL', () => {
    const img = fillerImage({ text: 'ExampleText', height: 900, width: 1600 })

    expect(img).toMatchSnapshot()
  })
})
