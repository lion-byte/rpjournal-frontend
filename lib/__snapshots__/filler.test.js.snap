// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filler Create filler image URL 1`] = `"https://via.placeholder.com/1600x900.png/535353/eeeeee?text=ExampleText"`;
